Clients: Update bin --help, add mapping for >=36.0 (#420)
rucio/documentation: docs/user/migrating_from_35_client.md

Reviewer: `migrating_to_36_client.md` might be a more accurate file name (this is not necessarily about migrating from 35, it could be a 34->36 migration)

| rucio list-file-replicas | rucio replica list |
| rucio list-dataset-replicas | rucio replica dataset |
| rucio add-dataset | rucio did add --type dataset |
| rucio add-container | rucio did add --type container |
| rucio attach | rucio did content add |
| rucio detach | rucio did content remove |
| rucio ls | rucio did list |
| rucio list-dids | rucio did list |
| rucio list-parent-dids | rucio did show --parent |
| rucio list-scopes | rucio scope list |
| rucio close | rucio did update --close |
| rucio reopen | rucio did update --open |
| rucio stat | rucio did show |
| rucio erase | rucio did remove |
| rucio list-content | rucio did content list |
| rucio list-content-history | rucio did content history |
| rucio upload | rucio upload |
| rucio get | rucio download |
| rucio download | rucio download |
| rucio get-metadata | rucio did metadata list |
| rucio set-metadata | rucio did metadata add |
| rucio delete-metadata | rucio did metadata remove |
| rucio list-rse-usage | rucio rse show |
| rucio list-account-usage | rucio account usage |
| rucio list-account-limits | rucio account limit list |
| rucio add-rule | rucio rule add |
| rucio delete-rule | rucio rule remove |

> | rucio rule-info | rucio rule info |

Reviewer: ```suggestion
| rucio rule-info | rucio rule show |
```

https://github.com/rucio/rucio/blob/master/lib/rucio/client/commands/rule.py#L34

---
title: Migrating from pre-36.0.0 Clients
---

> In Rucio 36.0.0, a new rucio client was released. 
> This new client redos the entire structure of both `rucio` and `rucio-admin`. 

Reviewer: ```suggestion
In Rucio 36.0.0, a new Rucio client was released, with breaking changes to the structure of both `rucio` and `rucio-admin`. 
```

> `rucio-admin` is made redundant by the update, their commands are included in the `rucio >=36.0` client. 

Reviewer: ```suggestion
`rucio-admin` is made redundant by the update, and its commands are included in the `rucio >=36.0` client. 
```

Note: 

>     Please report any issues you have with the new client [on this github issue](https://github.com/rucio/rucio/issues/7224)

Reviewer: ```suggestion
    Please report any issues you have with the new client [on this GitHub issue](https://github.com/rucio/rucio/issues/7224)
```

# Legacy Mapping

* Using `rucio-admin` or `rucio --legacy` automatically uses the old version. Old help menus are still visible with this method. 
* Old commands still work - they display a warning.

> * Not all commands are included in the 36 Client Release. Not all are one-to-one mapped. 

Reviewer: > Not all commands are included in the 36 Client Release

Could you clarify on this? I assume it means that not all commands have been migrated to the new client, but it could be interpreted to mean that some commands were left out of the 36 client release

Author: Yeah it's the first one. Maybe better wording would be 
```suggestion
* Not all commands are migrated to the new client structure. Not all included commands are one-to-one mapped. 
```

Reviewer: sounds good!

> * A warning it printed (not logged) to consoles to encourage migration, along with the closest mapped command. 

Reviewer: ```suggestion
* A warning is printed (not logged) to consoles to encourage migration, along with the closest mapped command. 
```
